tests/plugins/widget/integration/selection/manual/ieinputwidget.html

```diff
@@ -0,0 +1,43 @@
+<div id="editor">
+		<p>Add widget below.</p>
+	</div>
+
+	<script>
+		if ( !CKEDITOR.env.ie || CKEDITOR.env.version < 11) {
+			bender.ignore();
+		};
+
+		CKEDITOR.plugins.add( 'testplugin', {
+			requires: 'widget',
+			init: function( editor ) {
+				editor.widgets.add( 'testwidget', {
+					requiredContent: 'div(testwidget)',
+					allowedContent: 'div(testwidget)',
+					template: '<div class="testwidget">' +
+						'<input type="text">' +
+						'</div>',
+					button: 'Add widget',
+					upcast: function( element ) {
+						return element.name === 'div' && element.hasClass( 'testwidget' )
+					},
+					init: function() {
+						this.on( 'focus', function() {
+							var input = this.element.findOne( 'input' ),
+								range = this.editor.createRange();
+
+							input.focus( 1 );
+
+							range.selectNodeContents( input );
+							range.collapse();
+							range.select();
```

Comment:
That's crazy we need so many selection manipulations for simple input focus. But I suppose it's different issue where focus doesn't always mean selection change on IE?

Reply:
to have somehow working input and text area elements inside IE and Edge, we need to make such magic. :(